src/operator/tensor/dot-inl.h

```diff
@@ -791,17 +791,31 @@ inline void DotCsrDnsDnsImpl(const OpContext& ctx,
               s, num_threads, data_out.dptr<DType>());
         }
         num_threads = mxnet_op::get_num_threads<cpu>(data_out.shape_[0]);
+        bool dynamic = false;
+        if (data_out.shape_[0] > 1024 * 10) {
```

Comment:
Nit: could make "1024 * 10" a const as well and give a meaningful name like min_parallel_work_threshold.